feat(radarr): media management

Fuochi opened the pull request:
- add mediamanagement
- add naming
- add rootfolder missing methods

davidnewhall commented on radarr/mediamanagement.go:
```diff
@@ -0,0 +1,76 @@
+}
+
+// UpdateMediaManagementContext updates the mediaManagement.
+func (r *Radarr) UpdateMediaManagementContext(ctx context.Context, mMgt *MediaManagement) (*MediaManagement, error) {
```
maybe `input` instead of `mMgt`?

Fuochi replied:
I'm not a huge fan of non-meaningful input names, I used the same naming of sonarr. If you prefer, I can expand this as either `mediaManagement` or just `management`

davidnewhall replied:
it's fine as-is

davidnewhall commented on radarr/rootfolder.go:
```diff
@@ -1,29 +1,33 @@
 }
 
-// GetRootFolders returns all configured root folders.
+// GetRootFolders returns all configured rootFolders.
```
"root folders."

davidnewhall commented on radarr/mediamanagement_test.go:
```diff
@@ -0,0 +1,157 @@
+	tests := []*starr.TestMockData{
+		{
+			Name:           "200",
+			ExpectedPath:   path.Join("/", starr.API, radarr.APIver, "config/mediaManagement"),
```
```suggestion
			ExpectedPath:   path.Join("/", starr.API, radarr.APIver, "config", "mediaManagement"),
```

and the rest..?

davidnewhall commented on radarr/rootfolder.go:
```diff
@@ -1,29 +1,33 @@
 }
 
-// GetRootFoldersContext returns all configured root folders.
+// GetRootFoldersContext returns all configured rootFolders.
```
```suggestion
// GetRootFoldersContext returns all configured root folders.
```

Fuochi commented:
while testing this I noticed I've been to aggressive with the `omitempty` tag, especially for bool fields. I have to review this, I'll probably create a PR to fix across different resources

davidnewhall commented on radarr/rootfolder.go:
```diff
@@ -34,3 +38,57 @@ func (r *Radarr) GetRootFoldersContext(ctx context.Context) ([]*RootFolder, erro
 }
+
+// GetRootFolder returns a single root folder.
+func (r *Radarr) GetRootFolder(folderID int) (*RootFolder, error) {
```
folder ID is an int64

davidnewhall commented on radarr/rootfolder_test.go:
```diff
@@ -0,0 +1,226 @@
+			t.Parallel()
+			mockServer := test.GetMockServer(t)
+			client := radarr.New(starr.New("mockAPIkey", mockServer.URL, 0))
+			output, err := client.GetRootFolder(test.WithRequest.(int))
```
also int64

davidnewhall commented:
just a couple int->int64 fixes...

Fuochi commented:
> just a couple int->int64 fixes...

yes sorry, I was still looking at the `omitempty` tag, I'll try to fix this either today or tomorrow

Fuochi commented:
Just rebased, and fixed omitepty tags again, I thought I made a mess with the tags, but in the end only few fields were actually wrongly tagged (I updated the sonarr ones as well)

davidnewhall commented on radarr/rootfolder.go:
```diff
@@ -34,3 +38,57 @@ func (r *Radarr) GetRootFoldersContext(ctx context.Context) ([]*RootFolder, erro
+	}
+
+	return nil
+}
```
Are these file's changes intentional? As in, did you mean to add rootFolder changes in this pull request?

davidnewhall reviewed (approved):
This seems just fine. Thank you very much!
